### PR TITLE
Clean up eval-launcher scripts

### PR DESCRIPTION
**Pull request by amj**
A lot of refactoring, some constants, and the same codepath for cross-run and same-run matches.

### CODE REVIEW

File: cluster/evaluator/launch_eval.py [hunk 1]
```diff
@@ -16,37 +16,34 @@
...
-    jobname: string, appended to the container, used to differentiate the job names
-    (e.g. 'minigo-gpu-evaluator-v5-123-v7-456')
+    job_name: string, appended to the container, used to differentiate the job names
+    (e.g. 'minigo-cc-evaluator-v5-123-v7-456')
+    bucket_name: Where to write the sgfs, passed into the job as $BUCKET_NAME
+    completions: the number of completions desired
     """
-    if m1_path is None or m2_path is None or jobname is None:
-        print("Provide all of m1_path, m2_path, and jobname params")
+    if m1_path is None or m2_path is None or job_name is None or bucket_name is None:
```
Comment (sethtroisi) [inline cluster/evaluator/launch_eval.py]: consider:
all(var is not None for var in [m1_path, m2_path, job_name, bucket_name])
or
all([m1_path, m2_path, job_name, bucket_name])

File: cluster/evaluator/launch_eval.py [hunk 2]
```diff
@@ -184,13 +177,13 @@ def cleanup(api_instance=None):
...
-    pairs = make_pairs_for_model(model_num)
-    for p in pairs:
-        launch_eval(*p)
-
 
 def make_pairs_for_model(model_num=0):
+    """ Create a list of pairs of model nums; play every model nearby, then
+    every other model after that, then every fifth, etc.
+
+    Returns a list like [[N, N-1], [N, N-2], ... , [N, N-50]]
```
Comment (sethtroisi) [inline cluster/evaluator/launch_eval.py]: consider third example of [N, N-4]

File: cluster/evaluator/launch_eval.py (hunk 1 above)
Comment (amj) [reply cluster/evaluator/launch_eval.py]: Done.

File: cluster/evaluator/launch_eval.py (hunk 2 above)
Comment (amj) [reply cluster/evaluator/launch_eval.py]: Done.

**Comment by sethtroisi** [comment]
LGTM

**Pull request merged**